bittensor/_dataset/dataset_impl.py

```diff
@@ -473,25 +476,22 @@ def construct_text_corpus(self, min_data_len = 0):
                 i = 0
 
                 # --- Dont stop until the corpus size and the minimum data_length was reached.
-                for directory in directories:
-                    # --- Get a directory that leads to a datafile.
-                    random_datafile_dir = self.get_root_text_hash(directory)
-                    if random_datafile_dir == None:
-                        pass
-
-                    # --- Get text from the datafile directory
-                    text = self.get_text(random_datafile_dir)
-                    
-                    if text != None:
-                        text_list = text.split() 
-                        data_corpus.extend(text_list)
-                        total_dataset_size += int(random_datafile_dir['Size'])
-                        total_dataset_len += len(text_list)
-
-                    i += 1
-                    
-                    if (total_dataset_len > min_data_len) or self.IPFS_fails > self.IPFS_fails_max:
-                        break
+                with concurrent.futures.ThreadPoolExecutor(max_workers=cpu_count()) as executor:
+                    future_map = {}
+                    for idx, call_arg in enumerate(directories[:500]):
```

Comment:
Why the last 500 here?

Reply:
Horrible hack!

Basically, the first 500 directories (max) should have more than sufficient data for any reasonable query. 100 might also work. Didn't test it. However if you don't limit the amount you submit to the threadpool you hit some performance issues (bandwidth, and ram on smaller machines).